auto_submit/test/requests/check_pull_request_test.dart

```diff
@@ -2,47 +2,102 @@
     late CheckPullRequest checkPullRequest;
     late FakeConfig config;
+    late FakeGraphQLClient githubGraphQLClient;
     final FakeGithubService githubService = FakeGithubService();
     final FakePubSub pubsub = FakePubSub();
+    late PullRequestHelper prRequested1;
+    late PullRequestHelper prRequested2;
+    late List<QueryOptions> expectedOptions;
+    late QueryOptions flutterOption;
+    late QueryOptions cocoonOption;
     const String testTopic = 'test-topic';
-    const String login = "engine-flutter-autoroll";
+    const String rollorAuthor = "engine-flutter-autoroll";
     const String labelName = "warning: land on red to fix tree breakage";
-    const String repoName = 'cocoon';
-    const String autosubmitLabel = 'no_autosubmit';
+    const String cocoonRepo = 'cocoon';
+    const String noAutosubmitLabel = 'no_autosubmit';
+
+    setUp(() {
+      githubGraphQLClient = FakeGraphQLClient();
+      expectedOptions = <QueryOptions>[];
+
+      githubGraphQLClient.queryResultForOptions = (QueryOptions options) {
+        expect(options.variables['sOwner'], 'flutter');
+        final String? repoName = options.variables['sName'] as String?;
+        if (repoName == 'flutter') {
+          return createQueryResult(prRequested1);
+        } else if (repoName == 'cocoon') {
+          return createQueryResult(prRequested2);
+        } else {
+          fail('unexpected repo $repoName');
+        }
+      };
+
+      flutterOption = QueryOptions(
+        document: labeledPullRequestWithReviewsQuery,
+        fetchPolicy: FetchPolicy.noCache,
+        variables: <String, dynamic>{
+          'sOwner': 'flutter',
+          'sName': 'flutter',
+          'sPrNumber': 0,
+        },
+      );
+      cocoonOption = QueryOptions(
+        document: labeledPullRequestWithReviewsQuery,
+        fetchPolicy: FetchPolicy.noCache,
+        variables: <String, dynamic>{
+          'sOwner': 'flutter',
+          'sName': 'cocoon',
+          'sPrNumber': 1,
+        },
+      );
+    });
+
+    void _verifyQueries(List<QueryOptions> expectedOptions) {
+      githubGraphQLClient.verifyQueries(expectedOptions);
+    }
 
     test('Merges PR with successful status and checks', () async {
       final PullRequest pullRequest1 = generatePullRequest(prNumber: 0);
-      final PullRequest pullRequest2 = generatePullRequest(prNumber: 1, repoName: 'flutter', login: 'cocoon');
+      final PullRequest pullRequest2 = generatePullRequest(prNumber: 1, repoName: cocoonRepo);
       final List<PullRequest> pullRequests = <PullRequest>[pullRequest1, pullRequest2];
       for (PullRequest pr in pullRequests) {
         pubsub.publish(testTopic, pr);
       }
-      githubService.reviewsData = reviewsMock;
+
       githubService.checkRunsData = checkRunsMock;
-      githubService.repositoryStatusesData = repositoryStatusesMock;
-      githubService.commitData = commitMock;
       githubService.compareTowCommitsData = compareTowCommitsMock;
```

Comment:
Nit: compareTwoCommitsMock?

Reply:
Hi Keyong, do you mean that we should delete compareTwoCommitsMock here?

Reply:
Oh, no. Just a typo nit.

Reply:
Oh I see. No worries, I will add this back.

Reply:
Typo: `compareTowCommitsMock` -> `compareTwoCommitsMock`

Reply:
Oh I see Keyong, sorry I thought you mean that your nit is a typo. I will fix my typo here.